### PR TITLE
[dev-menu] update initial props on android and ios

### PR DESCRIPTION
**Pull request by ajsmth**
# Why

Some of the initial props passed to the dev menu react app are no longer used, some have been updated - this PR cleans up what is initially passed to the dev menu react app to include  props that are used by dev menu

Additionally, as multiple apps / manifests can be loaded throughout the lifetime of the dev-menu, its necessary to be able to update these values in React-land, it seems like the best way to do it is via the initial props

A second PR will achieve a similar API but for dev settings

# How

This required a small amount of refactoring but most of the functionality already was in place, and removal of some React code that called into native fns as it is no longer necessary

# Test Plan

Ensure the `initialProps` passed to the dev menu app includes the correct app info, and subsequent loads of different apps / manifests should, even updating manifest values and relaunching the app should render the correct appInfo in the dev menu

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-dev-menu/android/src/debug/java/expo/modules/devmenu/DevMenuAppInfo.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      }
+
+    return Bundle().apply {
+      putString("appVersion", appVersion)
+      putString("appName", appName)
+      putString("appIcon", appIcon)
+      putString("runtimeVersion", runtimeVersion)
+      putString("sdkVersion", sdkVersion)
+      putString("hostUrl", hostUrl)
+    }
```
Comment (lukmccall) [inline packages/expo-dev-menu/android/src/debug/java/expo/modules/devmenu/DevMenuAppInfo.kt:46]: Incorrect indentation

File: packages/expo-dev-menu/android/src/debug/java/expo/modules/devmenu/DevMenuAppInfo.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    val packageName = reactContext.packageName
+    val applicationInfo = packageManager.getApplicationInfo(packageName, 0)
+
+    if (applicationInfo.icon != null) {
+      appIcon = "" + applicationInfo.icon
+    }
+    //    TODO - figure out how to get resId for AdaptiveIconDrawable icons
+    return appIcon
+  }
+}
```
Comment (lukmccall) [inline packages/expo-dev-menu/android/src/debug/java/expo/modules/devmenu/DevMenuAppInfo.kt:68]:
```suggestion
}

```

**Review by lukmccall — APPROVED** [review]

File: packages/expo-dev-menu/ios/EXDevMenuAppInfo.m [hunk 3]
```diff
@@ -22,24 +23,30 @@ @implementation EXDevMenuAppInfo
...
   NSString *appName = [[NSBundle mainBundle] objectForInfoDictionaryKey: @"CFBundleDisplayName"] ?: [[NSBundle mainBundle] objectForInfoDictionaryKey: @"CFBundleExecutable"];
-  NSString *hostUrl = [bridge.bundleURL host] ?: @"";
-
-  if (manifest[@"name"] != nil) {
-    appName = manifest[@"name"];
-  }
   
-  if (manifest[@"version"] != nil) {
-    appVersion = manifest[@"version"];
+  DevMenuManager *manager = [DevMenuManager shared];
```
Comment (esamelson) [inline packages/expo-dev-menu/ios/EXDevMenuAppInfo.m:36]: keeping the manifest as a parameter (and having the caller be the one to pass in `DevMenuManager.shared.currentManifest`) might help keep this module more easily testable

Comment (ajsmth) [reply packages/expo-dev-menu/ios/EXDevMenuAppInfo.m:36]: hmm yea that's true - I would prefer to mock the devmenumanager singleton if possible and keep all this logic related to getting app info fields all in one place

**Comment by ajsmth** [comment]
commented out this test for now - have created a linear issue to reinstate the test but I'll need some help getting it setup locally: https://linear.app/expo/issue/ENG-4169/update-test-for-getappinfo

**Pull request merged**